### PR TITLE
feat(generate) : [CRRE-391] add confirm lightbox when generate orders to library

### PR DESCRIPTION
**Pull request by sjollois**
## Describe your changes
add confirm lightbox when generate orders to library
## Checklist tests
Generate orders to library from admin page. check with rejected orders
## Issue ticket number and link
[https://jira.support-ent.fr/browse/CRRE-391](url)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/i18n/fr.json [hunk 1]
```diff
@@ -815,5 +815,8 @@
   "Établissements généraux": "Générales et technologiques",
   "Établissements polyvalents": "Polyvalentes",
   "crre.students.number": "Nombre d'élèves",
-  "crre.basket.resubmit": "Panier soumis le "
+  "crre.basket.resubmit": "Panier soumis le ",
+  "crre.confirm.generate.library.title": "Confirmation de génération de demande libraire",
+  "crre.confirm.generate.library": "Confirmez-vous à envoyer ces commandes au libraire ?",
+  "crre.generate.refused.to.library": "Attention, votre commande comporte des lignes refusées."
```
Comment (qperie) [inline src/main/resources/i18n/fr.json:821]: A ajouter dans le en.json aussi tout ça

File: src/main/resources/public/ts/controllers/administrator/order/confirm-generate-library.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import {ng, template, toasts} from 'entcore';
+import {OrderRegion, OrdersRegion, Utils} from "../../../model";
+
+export const confirmGenerateLibraryController = ng.controller('confirmGenerateLibraryController',
+    ['$scope', ($scope) => {
+        $scope.generateLibraryOrder = async () => {
+            let selectedOrders;
```
Comment (qperie) [inline src/main/resources/public/ts/controllers/administrator/order/confirm-generate-library.ts:7]: Typage

File: src/main/resources/i18n/fr.json (hunk 1 above)
Comment (sjollois) [reply src/main/resources/i18n/fr.json:821]: il y a pas de en.json

File: src/main/resources/public/ts/controllers/administrator/order/confirm-generate-library.controller.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                if (data.status == 401){
+                    toasts.warning('crre.order.error.purse');
+                } else {
+                    toasts.warning('crre.order.region.library.create.err');
+                }
+                this.$scope.display.allOrdersSelected = false;
+                Behaviours.applicationsBehaviours['crre'].SnipletScrollService
+                    .sendScroll(true);
+            }
+        });
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/administrator/order/confirm-generate-library.controller.ts:64]: Si ton services retourne une 4XX ou 5XX alors cela est considérer comme une erreur. Donc tu ne passe pas dans le then. Mais tu peux faire ce traitement dans la catch
```suggestion
        await selectedOrders.generateLibraryOrder().then(async data => {
            toasts.confirm('crre.order.region.library.create.message');
            this.$scope.display.toggle = false;
            Utils.safeApply(this.$scope);
            this.$scope.display.allOrdersSelected = false;
            Behaviours.applicationsBehaviours['crre'].SnipletScrollService
                .sendScroll(true);
        }).catch((error: AxiosError) => {
            if (error.response.status == 401){
                toasts.warning('crre.order.error.purse');
            } else {
                toasts.warning('crre.order.region.library.create.err');
            }
            this.$scope.display.allOrdersSelected = false;
            Behaviours.applicationsBehaviours['crre'].SnipletScrollService
                .sendScroll(true);
        });
```

**Review by alicedraillard — APPROVED** [review]

File: src/main/resources/public/ts/directives/confirm-generate-library/confirm-generate-library.directive.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+interface ICrreScope extends IScope, ICrreProps {
+    vm: IViewModel;
+}
+
+interface ICrreProps {
+    display?: any;
+    projects?: Projects;
+
+    onCloseWaitingAdminLightbox?;
```
Comment (BastienBenard) [inline src/main/resources/public/ts/directives/confirm-generate-library/confirm-generate-library.directive.ts:23]: typage

File: src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts [hunk 5]
```diff
@@ -74,8 +74,23 @@ export const waitingOrderRegionController = ng.controller('waitingOrderRegionCon
             Utils.safeApply($scope);
         }
 
+        $scope.openConfirmGenerateLibraryLightbox = async () => {
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/administrator/order/waitingOrder.ts:77]: typage de retour

**Review by BastienBenard — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**